pkg/resources/manager/functions.go

```diff
@@ -27,6 +27,11 @@ func Allocate(machine *api.Machine, requiredResources core.ResourceList) error {
 	if len(requiredResources) == 0 {
 		return ErrResourcesEmpty
 	}
+
+	if mng.vmCountLimit != 0 && mng.vmCount >= mng.vmCountLimit {
```

Comment:
use it here can be dangerous because vmCount can be change with separate goroutine

Reply:
Moved it to manager